File: LinkedLists/LinkedList.js
let { Node } = require("./Node");
class LinkedList {
    constructor() {
        this.head = null;
        this.size = 0;
    }

    addToHead(value) {
        if (this.head == null) {
            this.head = new Node(value);
        } else {
            let node = new Node(value);
            node.next = this.head;
            this.head = node;
        }

        this.size++;
    }


    addToTail(value) {
        let node = this.head;
        while (node.next != null) {
            node = node.next;
        }

        let newNode = new Node(value);
        node.next = newNode;
        this.size++;
    }


}


let linkedList = new LinkedList();
linkedList.addToHead(12);
linkedList.addToTail(14);
for (let i = 0; i < 20; i++)
    linkedList.addToHead(i);
for (let i = 0; i < 20; i++) {
    linkedList.addToTail(i);
}
let startNode = linkedList.head;
while (startNode.next != null) {
    console.log(startNode.value);
    startNode = startNode.next;
}

console.log("size=" + linkedList.size);

File: Amazon/com/krishna/amazon/practice/LinkedList.java
package com.krishna.amazon.practice;

import java.util.HashSet;

import com.krishna.amazon.LinkNode;

public class LinkedList {
	private static LinkNode removeDups(LinkNode head) {
		LinkNode node = head;
		LinkNode previous = null;
		HashSet<Integer> hashSet = new HashSet<Integer>();
		while (node != null) {
			if (hashSet.contains(node.value)) {
				previous.next = node.next;
			} else {
				hashSet.add(node.value);
				previous = node;
			}
			node = node.next;
		}
		return node;
	}

	public static void main(String... args) {
		com.krishna.amazon.LinkedList linkedList = new com.krishna.amazon.LinkedList(new int[] { 12, 45, 78,45 });
		removeDups(linkedList.head);
		System.out.println(linkedList.toString());
	}
}
